### PR TITLE
Sites: show WP Admin link in sidebar to legacy users only (including VIP)

### PR DESCRIPTION
**Pull request by lancewillett**
This change refreshes the logic in sidebar for showing WP Admin link to legacy users.

To match WordPress.com front-end navigation, any user registered on WordPress.com before September 7, 2015 should see a "WP Admin" link at the bottom of their site's sidebar.

Removes the need to call the deprecated (and unused anywhere as far as I can see) user meta `flags` that are a holdover from Atlas (Newdash). I created a separate janitorial type PR could clean out the way that's pulled into the user data object, see #9765.

To test:

1. Log in with various users, bonus if you know the registration date of the user and can test 1 before and 1 after September 7, 2015.
2. If the user registered before that date, make sure the WP Admin links shows up at the bottom of the sidebar. screen shot 2016-12-01 at 11 13 40
3. If after, it shouldn't be there (except for WordPress.com VIP sites). screen shot 2016-12-01 at 11 13 50

### CODE REVIEW

**Comment by lancewillett** [comment]
CC @mtias for a code review. Relates to internal thread p58i-4pX-p2.

File: client/my-sites/sidebar/sidebar.jsx [hunk 1]
```diff
@@ -532,6 +522,26 @@ export class MySitesSidebar extends Component {
 		);
 	}
 
+	// Check for cases where WP Admin links should appear, where we need support for legacy reasons (VIP, older users, testing).
+	useWPAdminFlows() {
+		var site = this.getSelectedSite(),
```
Comment (mtias) [inline client/my-sites/sidebar/sidebar.jsx]: Probably change to `const` for each line.

**Comment by mtias** [comment]
I'm seeing the WP Admin link for my Jetpack site with these changes.

**Comment by lancewillett** [comment]
> I'm seeing the WP Admin link for my Jetpack site with these changes.

Is that a good side effect? Or should Jetpack sites ignore the link?

**Comment by mtias** [comment]
They should ignore it, I think. You are opting in to this interface.

**Comment by lancewillett** [comment]
@mtias Added a Jetpack check, thanks for catching that case.

File: client/my-sites/sidebar/sidebar.jsx (hunk 1 above)
Comment (lancewillett) [inline client/my-sites/sidebar/sidebar.jsx]: Done.

**Comment by enejb** [comment]
@mtias and @lancewillett it would be great to add a link for Jetpack sites as well. 
We are adding more and more links to Jetpack that open in calypso but there is no way to get back.

**Comment by lancewillett** [comment]
@enejb Please open a new issue or PR — this one is closed as it was WP.com only.

The reasoning against adding it for Jetpack is we haven't forced the new interface on Jetpack site owners, so don't need to give them a direct way back (which is different for legacy WP.com users).

**Comment by mtias** [comment]
> it would be great to add a link for Jetpack sites as well

I don't think so, at least not in the sidebar as a prominent navigation item. Individual flows that start in WP-Admin should have their own setup and way to get back. Missing functionality should have external links. @rickybanister @MichaelArestad

**Comment by rickybanister** [comment]
This is tricky as we're in a sort of liminal phase of supporting Jetpack sites on WordPress.com. The wp-admin link could be helpful in the near term, but I do agree that it can simply end up fracturing the experience. 

I think once we have a masterbar in Jetpack the loop is better closed. For now I think it's fine to continue to omit the wp-admin link. I think our larger struggle is getting folks to use Calypso, rather than users finding it difficult to get back to wp-admin.

**Comment by lancewillett** [comment]
Discussion continues in a new proposal: https://github.com/Automattic/wp-calypso/issues/11108

**Pull request merged**